src/Grids/grid_utils.jl

```diff
@@ -224,122 +217,103 @@ index_range_offset(::Colon, loc, topo, halo)          = - interior_parent_offset
 
 """
-    znodes(loc, grid, reshape=false)
+    znodes(grid, LX, LY, LZ, with_halos=false)
 
-Return a view over the interior `loc=Center` or `loc=Face` nodes
-on `grid` in the ``z``-direction. For `Bounded` directions,
-`Face` nodes include the boundary points.
-
-Keyword argument
-================
-- `reshape`: With `reshape=false` (default) the output is a 1D array while with 
-  `reshape=true` the output is a 3D array with size `1×1×Nz`.
-
-Examples
-========
+Return the positions over the interior nodes on `grid` in the ``z``-direction for the location `LX`,
+`LY`, `LZ`. For `Bounded` directions, `Face` nodes include the boundary points.
 
 ```jldoctest znodes
 julia> using Oceananigans
 
 julia> horz_periodic_grid = RectilinearGrid(size=(3, 3, 3), extent=(2π, 2π, 1), halo=(1, 1, 1),
-                                                 topology=(Periodic, Periodic, Bounded));
+                                            topology=(Periodic, Periodic, Bounded));
 
-julia> zC = znodes(Center, horz_periodic_grid)
+julia> zC = znodes(horz_periodic_grid, Center())
+3-element view(OffsetArray(::StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}, Int64}, 0:4), 1:3) with eltype Float64:
+ -0.8333333333333334
+ -0.5
+ -0.16666666666666666
+
+julia> zC = znodes(horz_periodic_grid, Center(), Center(), Center())
 3-element view(OffsetArray(::StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}, Int64}, 0:4), 1:3) with eltype Float64:
  -0.8333333333333334
  -0.5
  -0.16666666666666666
-```
 
-``` jldoctest znodes
-julia> zF = znodes(Face, horz_periodic_grid)
-4-element view(OffsetArray(::StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}, Int64}, 0:5), 1:4) with eltype Float64:
- -1.0
- -0.6666666666666666
- -0.3333333333333333
-  0.0
+julia> zC = znodes(horz_periodic_grid, Center(), Center(), Center(), with_halos=true)
+-1.1666666666666667:0.3333333333333333:0.16666666666666666 with indices 0:4
 ```
 """
-function znodes(loc, grid; reshape=false)
+@inline znodes(grid, LX::CellLocation, LY::CellLocation, LZ::CellLocation; kwargs...) = znodes(grid, LZ; kwargs...)
 
-    z = view(all_z_nodes(loc, grid),
-             interior_indices(loc, topology(grid, 3), grid.Nz))
 
-    return reshape ? Base.reshape(z, 1, 1, length(z)) : z
+"""
+    reshaped_xnodes(loc, grid; kwargs...)
+"""
+function reshaped_xnodes(grid, loc::CellLocation; kwargs...)
+    x = xnodes(grid, loc; kwargs...)
+    return Base.reshape(x, length(x), 1, 1)
 end
 
 """
-    nodes(loc, grid; reshape=false)
+    reshaped_ynodes(loc, grid; kwargs...)
+"""
+function reshaped_ynodes(grid, loc::CellLocation; kwargs...)
+    y = ynodes(grid, loc; kwargs...)
+    return Base.reshape(y, 1, length(y), 1)
+end
+
+"""
+    reshaped_znodes(loc, grid; kwargs...)
+"""
+function reshaped_znodes(grid, loc::CellLocation; kwargs...)
+    z = znodes(grid, loc; kwargs...)
+    return Base.reshape(z, 1, 1, length(z))
+end
```

Comment:
yeah we need generic code

Reply:
@glwagner we were discussing the need for these functions before, no? I couldn't find the thread exactly but basically I couldn't find a use for these functions and I think we should delete them.

You mentioned they were there because `Plots.jl` or some other plotting library needs arguments of a particular shape. I tried a few of these plotting libraries and found that they work just fine without these reshaped versions of `nodes`.

Reply:
I'm deleting them. If a user want to reshape they can do that as they wish :)

Reply:
great